external-crates/move/crates/move-compiler/src/parser/syntax.rs

```diff
@@ -2630,6 +2908,74 @@ fn parse_parameter(context: &mut Context) -> Result<(Mutability, Var, Type), Box
     Ok((mut_, v, t))
 }
 
+// (":" <Type>)?
+fn parse_ret_type(context: &mut Context, name: FunctionName) -> Result<Type, Box<Diagnostic>> {
+    if match_token(context.tokens, Tok::Colon)? {
+        parse_type(context)
+    } else {
+        Ok(sp(name.loc(), Type_::Unit))
+    }
+}
+
+fn parse_body(context: &mut Context, native: Option<Loc>) -> Result<FunctionBody, Box<Diagnostic>> {
+    match native {
+        Some(loc) => {
+            if let Err(diag) = consume_token(context.tokens, Tok::Semicolon) {
+                context.advance_until_stop_set(Some(*diag));
+            }
+            Ok(sp(loc, FunctionBody_::Native))
+        }
+        _ => {
+            let start_loc = context.tokens.start_loc();
+            let seq = if context.tokens.peek() == Tok::LBrace {
+                match consume_token(context.tokens, Tok::LBrace) {
+                    Ok(_) => parse_sequence(context)?,
+                    Err(diag) => {
+                        // error advancing past opening brace - assume sequence (likely first)
+                        // parsing problem and try skipping it
+                        advance_separated_items_error(
+                            context,
+                            Tok::LBrace,
+                            Tok::RBrace,
+                            /* separator */ Tok::Semicolon,
+                            /* for list */ true,
+                            *diag,
+                        );
+                        let _ = match_token(context.tokens, Tok::RBrace);
+                        (
+                            vec![],
+                            vec![],
+                            None,
+                            Box::new(Some(sp(
+                                context.tokens.current_token_loc(),
+                                Exp_::UnresolvedError,
+                            ))),
```

Comment:
Minor nit. I tend to find tuples a bit easier to read if they fit over one line. Binding the `Box::new(...)`  bit first should make that possible